### PR TITLE
Enable shellcheck errors.

### PR DESCRIPTION
**Pull request by mbobrovskyi**
#### What type of PR is this?
/kind cleanup

#### What this PR does / why we need it:
Enable SC2269, SC1091, SC2066, SC2086, SC2034, SC1083, SC2155, SC2046 shellcheck errors and resolve it.

#### Which issue(s) this PR fixes:

Fixes #

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

### CODE REVIEW

File: cmd/experimental/kjobctl/hack/update-codegen.sh [hunk 1]
```diff
@@ -19,21 +19,23 @@ set -o nounset
...
+CURRENT_DIR=$(dirname "${BASH_SOURCE[0]}")
+KJOBCTL_ROOT=$(realpath "$CURRENT_DIR/..")
 KJOBCTL_PKG="sigs.k8s.io/kueue/cmd/experimental/kjobctl"
-CODEGEN_PKG=$(cd $TOOLS_DIR && go mod download && $GO_CMD list -m -f "{{.Dir}}" k8s.io/code-generator)
+CODEGEN_PKG=$(cd "$TOOLS_DIR" && go mod download && $GO_CMD list -m -f "{{.Dir}}" k8s.io/code-generator)
 
-cd $(dirname ${BASH_SOURCE[0]})/..
+cd "$CURRENT_DIR/.."
 
+# shellcheck source=/dev/null
```
Comment (mimowo) [inline cmd/experimental/kjobctl/hack/update-codegen.sh:29]: Is it needed? From https://www.shellcheck.net/wiki/SC1090: 
ShellCheck v0.7.2+ will strip a single expansion followed by a slash, e.g. ${var}/util.sh or $(dirname "${BASH_SOURCE[0]}")/util.sh, and treat them as ./util.sh. This allows the use of source-path directives or -P flags to specify the location.

File: .shellcheckrc [hunk 2]
```diff
@@ -5,31 +5,3 @@
...
-disable=SC2269
-# Reasons include: file not found, no permissions, not included on the command line, not allowing shellcheck to follow files with -x, etc.
-# https://github.com/koalaman/shellcheck/wiki/SC1091
-disable=SC1091
-# Since you double quoted this, it will not word split, and the loop will only run once
-# https://github.com/koalaman/shellcheck/wiki/SC2066
-disable=SC2066
-# Double quote to prevent globbing and word splitting
-# https://github.com/koalaman/shellcheck/wiki/SC2086
-disable=SC2086
```
Comment (mimowo) [inline .shellcheckrc:23]: I see this disabled in many places in the code. It would be less code to disable it here, and this is where people will look for configuration. Are there any benefits from enabling it here, and disabling in code?

File: hack/update-codegen.sh [hunk 3]
```diff
@@ -19,10 +19,12 @@ set -o nounset
...
 
 GO_CMD=${1:-go}
-KUEUE_ROOT=$(realpath $(dirname ${BASH_SOURCE[0]})/..)
+CURRENT_DIR=$(dirname "${BASH_SOURCE[0]}")
+KUEUE_ROOT=$(realpath "${CURRENT_DIR}/..")
 CODEGEN_PKG=$($GO_CMD list -m -mod=readonly -f "{{.Dir}}" k8s.io/code-generator)
-cd $(dirname ${BASH_SOURCE[0]})/..
+cd "${CURRENT_DIR}/.."
 
+# shellcheck source=/dev/null
```
Comment (mimowo) [inline hack/update-codegen.sh:27]: Same question here - why is this needed?

File: .shellcheckrc (hunk 2 above)
Comment (mbobrovskyi) [reply .shellcheckrc:23]: Disabled only on places where we expect this situation. For example [here](https://github.com/kubernetes-sigs/kueue/blob/7f35d8da425ca8c35ea14a4dc982f48fa0d9ef02/hack/e2e-test.sh#L70) we should split `$GINKGO_ARGS` by spaces and allow user to put multiple args in variables. In another places (look at [commit](https://github.com/kubernetes-sigs/kueue/pull/2763/commits/7f35d8da425ca8c35ea14a4dc982f48fa0d9ef02)) we should prevent this situation.

File: cmd/experimental/kjobctl/hack/update-codegen.sh (hunk 1 above)
Comment (mbobrovskyi) [reply cmd/experimental/kjobctl/hack/update-codegen.sh:29]: Yes, without this directive I get an error:
```
In ./cmd/experimental/kjobctl/hack/update-codegen.sh line 29:
source "${CODEGEN_PKG}/kube_codegen.sh"
       ^-- SC1091 (info): Not following: ./kube_codegen.sh: openBinaryFile: does not exist (No such file or directory)
```

We don't know exactly what the place we will have on `${CODEGEN_PKG}`.

File: .shellcheckrc (hunk 2 above)
Comment (mimowo) [reply .shellcheckrc:23]: Thanks, this makes sense - so there are many more places which benefit from enabling it.

File: cmd/experimental/kjobctl/hack/update-codegen.sh (hunk 1 above)
Comment (mimowo) [reply cmd/experimental/kjobctl/hack/update-codegen.sh:29]: Nice, thanks for sharing the error output.

**Comment by mimowo** [comment]
/lgtm
/approve
Thanks

**Comment by tenzen-y** [comment]
Fixes: https://github.com/kubernetes-sigs/kueue/issues/1795
